server_extensions_api/src/main/java/com/generalbytes/batm/server/extensions/IExtensionContext.java

```diff
@@ -493,4 +493,21 @@ IIdentity updateIdentity(String identityId, String externalId, int state, int ty
      */
     IApiAccess getAPIAccessByKey(String apiKey);
 
+    /**
+     * Creates Identity-linked Discount
+     *
+     * @param identityPublicId Public ID of an existing identity to be linked.
+     * @param code Defined code to be used (upper-cased) or null for code to be generated.
+     * @param buyDiscount
+     * @param sellDiscount
+     * @param validityFrom
+     * @param validityTill
+     * @param maximumUsages
+     * @param maximumUsagesPerIdentity
+     * @param minimumTransactionAmount
+     * @param totalLimit
+     * @param fiatCurrency in which currency are limit amounts
+     * @return Newly created Discount details.
+     */
+    IDiscount createIdentityLinkedDiscount(String identityPublicId, String code, BigDecimal buyDiscount, BigDecimal sellDiscount, Date validityFrom, Date validityTill, Long maximumUsages, Long maximumUsagesPerIdentity, BigDecimal minimumTransactionAmount, BigDecimal totalLimit, String fiatCurrency);
```

Comment:
1. rename this method to createDiscount.
2. make String identityPublicId optional and move it as a last parameter.